### PR TITLE
Start setup of pysmurf_aggregator from stream listener

### PR DESCRIPTION
**Pull request by BrianJKoopman**
Use old stream listener as basis for the pysmurf aggregator. This makes minimal
modifications for allowing the listener to run in a container. We also create
the Agent's Dockerfile and add it to the docker-compose file.

Just wanted to get a working baseline to start from. Open to suggestions for naming, etc.

### CODE REVIEW

File: agents/pysmurf_aggregator/stream_listener.py [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+        while self.is_streaming:
+            if writer is None:
+                start_time = time.time()
+                filepath = _create_file_path(start_time, data_dir)
+                writer = core.G3Writer(filename=filepath)
+
+                if last_meta is not None:
+                    writer(last_meta)
+
+            frames = reader.Process(None)
```
Comment (jlashner) [inline agents/pysmurf_aggregator/stream_listener.py]: I don't know if this is actually safe to do in the process loop. If the writer stops writing frames it'll freeze here, and you won't be able to cleanly exit. We should probably have a function run in a separate thread that loops while `self.is_streaming` is true and calls `reader.Process`, and adds frames to a queue. 

I think if the NetworkSender is destroyed, reader.Process will return, but its not always destroyed immediately if it's created in python. This has definitely been a problem for me in the past when testing the stream-simulator with the stream-listener.

File: agents/pysmurf_aggregator/stream_listener.py [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+        self.log = self.agent.log
+        self.address = address
+        self.port = port
+        self.is_streaming = False
+
+    def start_stream(self, session, params=None):
+        if params is None:
+            params = {}
+
+        time_per_file = params.get("time_per_file", 60*60)  # [sec]
```
[the remark this answers is not in the document]
Comment (BrianJKoopman) [reply agents/pysmurf_aggregator/stream_listener.py]: Ah, yup, did that already, sorry have been actively working on this right now, was avoiding pushing repeatedly. I made this an data-dir a site-argument, as well as an auto-start argument so this agent can just startup its only process right away.

File: agents/pysmurf_aggregator/stream_listener.py [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+        data_dir = params.get("data_dir", "data/")
```
[the remark this answers is not in the document]
Comment (BrianJKoopman) [reply agents/pysmurf_aggregator/stream_listener.py]: I made the default `/data/` when I moved this to a site-argument. I'm not sure it makes any less sense in a docker container than `data/` would outside of the container...

**Comment by jlashner** [comment]
I think this looks pretty good. Comments about time_per_file being a command line arg and data_dir being a required arg aren't super necessary for this PR, I just thought I would point them out. 

I think it's pretty important to move reader.Process from the aggregation loop, because I'm pretty sure this will cause problems. 

Also I think we should change the name from pysmurf_aggregator, since the detector timestreams aren't coming from pysmurf. I'd recommend something like tod_aggregator or timestream_aggregator.

File: agents/pysmurf_aggregator/stream_listener.py (hunk 1 above)
Comment (BrianJKoopman) [reply agents/pysmurf_aggregator/stream_listener.py]: Right, so it does freeze if the streamer stops sending it data and you then ask the aggregator to exit.

For more context, `reader.Process()` will hang until it either receives a frame or the NetworkSender is destroyed.

Reiterating our conversation on Slack for others, I'm not sure moving this to a different thread is the right solution, as then the problem is that separate thread hangs and can't properly exit. I'm not sure the proper solution to this. I'll think about it over the weekend.

**Comment by BrianJKoopman** [comment]
Alright, I went with "timestream_aggregator". I've also made a page for the Sphinx docs with example ocs and Docker configuration sections. I think we're ready to merge if that's alright. I'll open an issue about the `reader.Process` bit once we do.

**Comment by BrianJKoopman** [comment]
Realized I missed renaming within the docker-compose file at the top of the repo that builds all the images. Renamed and updated.

**Pull request merged**